CONTRIBUTING.md

```diff
@@ -19,422 +19,442 @@ under the License.
+```
 
-We currently rely heavily on Travis CI for running the full Airflow test suite
-as running all of the tests locally requires significant setup.  You can setup
-Travis CI in your fork of Airflow by following the
-[Travis CI Getting Started guide][travis-ci-getting-started].
+This generates a file in `airflow/migrations/versions/` which contains the changes to make to the schema. In 
+that file, implement the changes in the `upgrade()` and `downgrade()` functions for respectively upgrading and 
+downgrading the schema.
 
-There are two different options available for running Travis CI which are
-setup as separate components on GitHub:
+You can inspect the entire history of schema changes with `alembic history`:
 
-1. **Travis CI GitHub App** (new version)
-1. **Travis CI GitHub Services** (legacy version)
+```
+$ alembic history
+021ae05a82b6 -> 2da817564f87 (head), add new column to table
+6e96a59344a4, 004c1210f153 -> 021ae05a82b6 (mergepoint), empty message
+939bb1e647c8 -> 6e96a59344a4, Make TaskInstance.pool not nullable
+939bb1e647c8 -> 004c1210f153, increase queue name size limit
+...
+```
 
-#### Travis CI GitHub App (new version)
+## Conventions
 
-1. Once installed, you can configure the Travis CI GitHub App at
-https://github.com/settings/installations -> Configure Travis CI.
+- We apply a line length of 110 characters. For IntelliJ/PyCharm, you can configure to display a vertical line
+at 110 characters in Preferences -> Editor -> Code Style -> Hard Wrap at -> set to 110.
+- We apply the reStructuredText docstring format.
 
-1. For the Travis CI GitHub App, you can set repository access to either "All
-repositories" for convenience, or "Only select repositories" and choose
-`<username>/airflow` in the dropdown.
+# Pull request guidelines
 
-1. You can access Travis CI for your fork at
-`https://travis-ci.com/<username>/airflow`.
+Before you submit a pull request from your forked repo, check that it meets these guidelines:
 
-#### Travis CI GitHub Services (legacy version)
+- Please make sure a PR touches only a single feature. So e.g. when adding a new argument to an operator, add 
+only that one argument and don't fix a typo in a different argument. Unrelated changes are confusing to a 
+reviewer and make it harder to trace changes for others and your future self. We rather have 2 PRs touching 2 
+features, than a single PR changing multiple features.
+- Every PR, except for documentation changes, requires an associated JIRA issue.
+- Every commit subject, except for documentation changes, must start with "[AIRFLOW-1234]" where 1234 references the related JIRA issue.
+- Documentation only changes do not require a JIRA issue and the commit subject can start with "[AIRFLOW-XXX]".
+- The PR title must also start with [AIRFLOW-XXX] or [AIRFLOW-1234].
+- Every PR should consist of a single commit. Please [rebase your fork](http://stackoverflow.com/a/7244456/1110993) and squash commits.
```

Comment:
i think this is no longer enforced. Does it actual requirement to always squash commits for code review? Even commiters don't do that in their PRs

Reply:
Agree, though rebasing should still always be required.

Reply:
It should  not be strictly needed for reviews but I think it should be encouraged when the final "OK" is given to squash the commits.  I think we should encourage contributors to use the `--fixup` commit flag to apply fixes. Although I find it really great if contributor performs `git rebase --autosquash` before the final merge. It often happens now that committer performs "squash&merge" via github interface when merging, but sometimes I can see that one PR can contain several commits (even with the same AIRFLOW-XXX) and in this case it should be mentioned by the committer that it should not be squashed any further.